test/igetest.c

```diff
@@ -145,129 +151,113 @@ static struct bi_ige_test const bi_ige_test_vectors[] = {
-            printf("IGE test vector %d failed (with in == out)\n", n);
-            hexdump(stdout, "key", v->key, sizeof v->key);
-            hexdump(stdout, "iv", v->iv, sizeof v->iv);
-            hexdump(stdout, "in", v->in, v->length);
-            hexdump(stdout, "expected", v->out, v->length);
-            hexdump(stdout, "got", buf, v->length);
-
-            ++errs;
-        }
+    const struct ige_test *const v = &ige_test_vectors[n];
+    AES_KEY key;
+    unsigned char buf[MAX_VECTOR_SIZE];
+    unsigned char iv[AES_BLOCK_SIZE * 2];
+    int testresult = 1;
+
+    if (!TEST_int_ge(BIG_TEST_SIZE, TEST_SIZE)
+        || !TEST_int_le(v->length, MAX_VECTOR_SIZE))
+        return 0;
+
+    if (v->encrypt == AES_ENCRYPT)
+        AES_set_encrypt_key(v->key, 8 * sizeof v->key, &key);
+    else
+        AES_set_decrypt_key(v->key, 8 * sizeof v->key, &key);
+    memcpy(iv, v->iv, sizeof iv);
+    AES_ige_encrypt(v->in, buf, v->length, &key, iv, v->encrypt);
+
+    if (!TEST_mem_eq(v->out, v->length, buf, v->length)) {
+        TEST_info("IGE test vector %d failed", n);
+        hexdump(stderr, "key", v->key, sizeof v->key);
+        hexdump(stderr, "iv", v->iv, sizeof v->iv);
+        hexdump(stderr, "in", v->in, v->length);
+        testresult = 0;
     }
 
-    for (n = 0; n < OSSL_NELEM(bi_ige_test_vectors); ++n) {
-        const struct bi_ige_test *const v = &bi_ige_test_vectors[n];
-        AES_KEY key1;
-        AES_KEY key2;
-        unsigned char buf[MAX_VECTOR_SIZE];
-
-        assert(v->length <= MAX_VECTOR_SIZE);
-
-        if (v->encrypt == AES_ENCRYPT) {
-            AES_set_encrypt_key(v->key1, 8 * v->keysize, &key1);
-            AES_set_encrypt_key(v->key2, 8 * v->keysize, &key2);
-        } else {
-            AES_set_decrypt_key(v->key1, 8 * v->keysize, &key1);
-            AES_set_decrypt_key(v->key2, 8 * v->keysize, &key2);
-        }
-
-        AES_bi_ige_encrypt(v->in, buf, v->length, &key1, &key2, v->iv,
-                           v->encrypt);
-
-        if (memcmp(v->out, buf, v->length)) {
-            printf("Bidirectional IGE test vector %d failed\n", n);
-            hexdump(stdout, "key 1", v->key1, sizeof v->key1);
-            hexdump(stdout, "key 2", v->key2, sizeof v->key2);
-            hexdump(stdout, "iv", v->iv, sizeof v->iv);
-            hexdump(stdout, "in", v->in, v->length);
-            hexdump(stdout, "expected", v->out, v->length);
-            hexdump(stdout, "got", buf, v->length);
-
-            ++errs;
-        }
+    /* try with in == out */
+    memcpy(iv, v->iv, sizeof iv);
+    memcpy(buf, v->in, v->length);
+    AES_ige_encrypt(buf, buf, v->length, &key, iv, v->encrypt);
+
+    if (!TEST_mem_eq(v->out, v->length, buf, v->length)) {
+        TEST_info("IGE test vector %d failed (with in == out)", n);
+        hexdump(stderr, "key", v->key, sizeof v->key);
+        hexdump(stderr, "iv", v->iv, sizeof v->iv);
+        hexdump(stderr, "in", v->in, v->length);
```

Comment:
Yes, we should.  The mem compare routines include this, it should be made exportable in due course.